contracts/docs/scripts/create-docs-website-pr-branch.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+### PURPOSE
+# This script will automate pushing a PR branch into the docs.linea.build repo
+# The new PR branch will contain autogenerated smart contract documentation for the Linea monorepo
+# You must manually create the Github PR yourself (from the new branch)
+
+### ASSUMPTIONS
+# - Requires permissions to create a branch on docs.linea.build repo
+# - Must execute this script from within linea-monorepo (can be anywhere)
+
+### CONSTANTS
+DOCS_WEBSITE_REPO_NAME=doc.linea
+DOCS_WEBSITE_REPO_GIT_LINK=https://github.com/Consensys/$DOCS_WEBSITE_REPO_NAME.git
+DOCS_REPO_PR_BRANCH_NAME=update-linea-smart-contract-docs
+DOCS_REPO_PR_COMMIT_MESSAGE="Update Linea smart contract docs"
+
+UPDATE_SIDEBAR_SCRIPT_NAME=updateSidebar.js
+UPDATE_SIDEBAR_SCRIPT_PATH=contracts/docs/scripts/$UPDATE_SIDEBAR_SCRIPT_NAME
+MONOREPO_SMART_CONTRACT_DOCS_DIRECTORY=contracts/docs/api
+
+DOCS_REPO_SMART_CONTRACT_DOC_DIRECTORY=docs/api/linea-smart-contracts
+
+MAX_FOLDER_DEPTH=3
+
+### SCRIPT START
+
+# Ensure that `contracts` is the starting working directory
+MONOREPO_ROOT_PATH=$(git rev-parse --show-toplevel)
+cd $MONOREPO_ROOT_PATH
+cd contracts
+
+# Docgen
+npx hardhat docgen
```

Comment:
do we need a step beforehand for `pnpm install` to be sure all dependencies are installed? same with Foundry - e.g. I have just pulled the repo and want the docs, how does it work cleanly?

Reply:
Good point, have added a section in this script to check for installations of hardhat and foundry